### PR TITLE
Update musl to 1.1.22

### PR DESCRIPTION
**Pull request by indygreg**
This is the latest available version. I noticed Rust wasn't using the
latest version when I attempted to have Cargo link object files produced
outside of Cargo / Rust's toolchain and was getting missing symbol
errors. Those missing symbols were added in 1.1.21 and 1.1.22.

I'm not fully sure of the implications of this change or how
comprehensive the test coverage is. Upstream changes in 1.1.21 and
1.1.22 can be found at
https://git.musl-libc.org/cgit/musl/tree/WHATSNEW?h=v1.1.22&id=e97681d6f2c44bf5fa9ecdd30607cb63c780062e#n1989.

### CODE REVIEW

File: src/ci/docker/scripts/musl-toolchain.sh [hunk 1]
```diff
@@ -3,7 +3,7 @@
 #
 # Versions of the toolchain components are configurable in `musl-cross-make/Makefile` and
 # musl unlike GLIBC is forward compatible so upgrading it shouldn't break old distributions.
-# Right now we have: Binutils 2.27, GCC 6.3.0, musl 1.1.18
+# Right now we have: Binutils 2.27, GCC 6.3.0, musl 1.1.22
```
Comment (sanxiyn) [inline src/ci/docker/scripts/musl-toolchain.sh]: This seems wrong. Looking at https://github.com/richfelker/musl-cross-make/blob/v0.9.7/Makefile, this still uses musl 1.1.18. Maybe you want to update to v0.9.8, or do this as another PR?

**Comment by lzutao** [comment]
cc @mati865 in case you interested.

**Comment by mati865** [comment]
I saw it but `sanxiyn` already explained what is wrong.

**Comment by indygreg** [comment]
I saw that reference to musl 1.1.18 and figured it was missed when musl was last updated from 1.1.18 to 1.1.20. But I see now it belongs to an external project.

I may very well submit another PR to get musl-cross-make bumped to the latest/greatest versions.

**Comment by sanxiyn** [comment]
As far as I can tell, this PR is harmless but won't be useful for you, because dist-x86_64-musl does not use musl.sh at all. It uses musl-toolchain.sh.

**Comment by sanxiyn** [comment]
Note: another PR was #61300 and it got merged.

**Comment by indygreg** [comment]
Gentle ping on review.
